src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphWorkspaceDAO.scala

```diff
@@ -56,4 +56,18 @@ class GraphWorkspaceDAO extends WorkspaceDAO with GraphDAO {
       case None => false
     }
   }
+
+  val LOCK_PROPERTY = "isLocked"
+
+  override def isLocked(workspaceContext: WorkspaceContext): Boolean = {
+    workspaceContext.workspaceVertex.getProperty(LOCK_PROPERTY)
+  }
+
+  override def lock(workspaceContext: WorkspaceContext): Unit = {
+    workspaceContext.workspaceVertex.setProperty(LOCK_PROPERTY,true)
+  }
+
+  override def unlock(workspaceContext: WorkspaceContext): Unit = {
+    workspaceContext.workspaceVertex.setProperty(LOCK_PROPERTY,false)
+  }
```

Comment:
Yes, but in order for the field to show up on the Context you'd have to
pull it out of the vertex manually like we do with bucketName.

Workspaces are a bit of a special case in that they have multiple scala
representations, none of which fully reflect the vertex contents. I'm okay
with this.
On Sep 4, 2015 9:13 AM, "dvoet" <notifications@github.com> wrote:

In
src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphWorkspaceDAO.scala
<https://github.com/broadinstitute/rawls/pull/113#discussion_r38749756>:

> @@ -56,4 +56,18 @@ class GraphWorkspaceDAO extends WorkspaceDAO with GraphDAO {
>        case None => false
>      }
>    }
> +
> +  val LOCK_PROPERTY = "isLocked"
> +
> +  override def isLocked(workspaceContext: WorkspaceContext): Boolean = {
> +    workspaceContext.workspaceVertex.getProperty(LOCK_PROPERTY)
> +  }
> +
> +  override def lock(workspaceContext: WorkspaceContext): Unit = {
> +    workspaceContext.workspaceVertex.setProperty(LOCK_PROPERTY,true)
> +  }
> +
> +  override def unlock(workspaceContext: WorkspaceContext): Unit = {
> +    workspaceContext.workspaceVertex.setProperty(LOCK_PROPERTY,false)
> +  }

why do we need special functions for this? shouldn't we just change the
property on the Workspace and call save?

—
Reply to this email directly or view it on GitHub
<https://github.com/broadinstitute/rawls/pull/113/files#r38749756>.